alerting_notification_policy.go

```diff
@@ -118,3 +118,7 @@ func (c *Client) SetNotificationPolicy(np *NotificationPolicy) error {
 	}
 	return c.request("PUT", "/api/v1/provisioning/policies", nil, bytes.NewBuffer(req), nil)
 }
+
+func (c *Client) ResetNotificationPolicy() error {
```

Comment:
I wouldn't go with `DeleteNotificationPolicy`. I'm fine with any of the other two options :)

I don't see any identifiers, does this method reset all notification policies? If so, wouldn't it be better to use the plural form "policies" instead of "policy"?

Reply:
It resets the entire tree of notification policies. So, it's really ambiguous english-wise... is it changing many objects (each node of the tree) or is it changing just one (the tree itself)?

I like keeping it singular because `SetNotificationPolicy` and `GetNotificationPolicy` operates on a single object and not a slice. It's consistent, IMO, `GetX` should return `X` and `GetXs` should return `[]X`.

Thoughts on using the term `PolicyTree` as a disambiguator? `ResetNotificationPolicyTree` does not have this problem.

Reply:
I love `ResetNotificationPolicyTree` actually! Not a dealbreaker though.

Reply:
Renamed to `Tree` everywhere. This is safe to do because the other methods have not been released yet.